### PR TITLE
Fixed issues #32, #40, #41, #50

### LINKED ISSUE #32: Panic if return value is used in an expression
```
package main

func foob()(ret bool) {
    ret = true
}

func fooi()(ret i32) {
    ret = 5
}

func main()() {
    var b bool = foob() == true
    var i i32 = fooi() + 5
    if fooi() < 6 {}
    if foob() == true {}
}
```
output:
```
panic: runtime error: index out of range

goroutine 1 [running, locked to thread]:
github.com/skycoin/cx/cxgo/actions.ArithmeticOperation(0xc4203c6078, 0x1, 0x1, 0xc4203c6088, 0x1, 0x1, 0xc4201244d0, 0xc4200af3a0, 0xbaa6e658, 0xf813e7b5bf0ba804)
	/home/reash/go/src/github.com/skycoin/cx/cxgo/actions/cxgo_compiled.go:1036 +0x10de
github.com/skycoin/cx/cxgo/actions.ShorthandExpression(0xc4203c6078, 0x1, 0x1, 0xc4203c6088, 0x1, 0x1, 0x0, 0x1, 0x1, 0x1)
	/home/reash/go/src/github.com/skycoin/cx/cxgo/actions/cxgo_compiled.go:934 +0xa3
main.yyParse(0x9486e0, 0xc4203c0000, 0xc4203c0000)
	/home/reash/go/src/github.com/skycoin/cx/cxgo/cxgo.go:2136 +0x768d
main.main()
	/home/reash/go/src/github.com/skycoin/cx/cxgo/cxgo.nn.go:8369 +0x1470

```

### PR DESCRIPTION
**Pull request by amherag**
Fixes #32, #40, #41, #50 

Changes:
- Fixed the issues.

Does this change need to be mentioned in CHANGELOG.md?
Yes.

### CODE REVIEW

File: cx/cxprogram.go [hunk 1]
```diff
@@ -32,6 +32,13 @@ type CXProgram struct {
 	Terminated     bool
 }
 
+// CXCall ...
```
Comment (ingwal) [inline cx/cxprogram.go]: This would make more sense near CXFunction, wouldn't it?  But we can fix that later if we agree.

File: cx/makers.go [hunk 2]
```diff
@@ -77,23 +77,26 @@ func MakeArgument(name string, fileName string, fileLine int) *CXArgument {
 }
 
 // MakeNative ...
-func MakeNative(opCode int, inputs []int, outputs []int) *CXFunction {
+func MakeNative(opCode int, inputs, outputs []*CXArgument) *CXFunction {
```
Comment (ingwal) [inline cx/makers.go]: I'm not very fond of this style.  I think that both 'inputs' and 'outputs' should have an explicit type declaration.

File: cx/opcodes_bare.go [hunk 3]
```diff
@@ -270,206 +270,581 @@ func DumpOpCodes(opCode int) () {
...
-	AddOpCode(OP_STR_F64, "str.f64", []int{TYPE_STR}, []int{TYPE_F64})
-
-	AddOpCode(OP_APPEND, "append", []int{TYPE_UNDEFINED, TYPE_UNDEFINED}, []int{TYPE_UNDEFINED})
-	AddOpCode(OP_ASSERT, "assert", []int{TYPE_UNDEFINED, TYPE_UNDEFINED, TYPE_STR}, []int{TYPE_BOOL})
-	AddOpCode(OP_TEST, "test", []int{TYPE_UNDEFINED, TYPE_UNDEFINED, TYPE_STR}, []int{})
-	AddOpCode(OP_PANIC, "panic", []int{TYPE_UNDEFINED, TYPE_UNDEFINED, TYPE_STR}, []int{})
+	AddOpCode(OP_IDENTITY, "identity",
+		[]*CXArgument{newOpPar(TYPE_UNDEFINED, false)},
+		[]*CXArgument{newOpPar(TYPE_UNDEFINED, false)})
+	AddOpCode(OP_JMP, "jmp",
```
Comment (ingwal) [inline cx/opcodes_bare.go]: I assume that this style change was demanded by some automatic tool.  It's not an improvement in this case...

Comment (amherag) [reply cx/opcodes_bare.go]: the lines were too long

File: cx/makers.go (hunk 2 above)
Comment (amherag) [reply cx/makers.go]: they both kinda have now with CXArgument. or what do you mean?

**Pull request merged**